File: VagueList.py
import random

class VagueList:
    def __init__(self, cont):
        self.cont = cont

    def __getitem__(self, index):
        return self.cont[(index + random.randint(-1, 1)) % len(self.cont)]

    def __len__(self):
        return random.randint(0, len(self.cont)*2)

vague_list = VagueList(["A", "B", "C", "D", "E"])
print(len(vague_list))
print(len(vague_list))
print(vague_list[4])
print(vague_list[4])

File: src/com/andrewfromfarnorth/VagueList.java
package com.andrewfromfarnorth;

import java.util.Random;

public class VagueList {
    private int[] cont;

    public VagueList() {
        System.out.println("Test");
    }
    VagueList(int[] cont) {
        this.cont = cont;
    }

    int getLength() {
        final Random random = new Random();
        final int randomInt = random.nextInt(cont.length);
        final int i = randomInt * 2;
        return i;
    }

    int getMember(int index) {
        final Random random = new Random();
        final int randomInt = random.nextInt(3) - 1;
        final int result = cont[(index + randomInt) % cont.length] ;
        return result;
    }

}
